Jetpack Connect: Add skip button to site type, site topic and user type steps

tyxla opened the pull request:
#### Changes proposed in this Pull Request

* Introduce a new `<SkipButton />` component.
* Add a skip button to the Site Type step.
* Add a skip button to the Site Topic step.
* Add a skip button to the User Type step.

#### Preview
Site type step:
![](https://cldup.com/0LA1GZBoYs.png)

Site topic/vertical step:
![](https://cldup.com/SE5xy5jCZt.png)

User type step:
![](https://cldup.com/O-gywMJAn5.png)

#### Testing instructions

* Checkout this branch, or spin it up at calypso.live.
* Create a new JN site with the latest Jetpack.
* Connect the site (adding `&calypso_env=development` to the connection URL if you use local calypso) and reach the site type step.
* Verify skip button on the site type step leads you to the site topic step.
* Verify skip button on the site topic/vertical step leads you to the user type page for the Jetpack bleeding edge, or plans page for older Jetpacks.
* Verify skip button on the user type step leads you to the plans page.
* Verify selecting a site type and continuing still works well.
* Verify selecting a site topic/vertical and continuing still works well.
* Verify selecting a user type and continuing still works well.

We're also recording some tracks events, in case you wish to test those:
* Type `localStorage.setItem( 'debug', 'calypso:analytics:tracks' )` in your browser console.
* On the site type step, click the skip button.
* Verify `calypso_jpc_skip_button_click` event gets recorded (our generic skip button click event).
* Verify `calypso_jpc_skipped_site_type` event gets recorded (our specific site type skip event).
* On the site topic step, click the skip button.
* Verify `calypso_jpc_skip_button_click` event gets recorded (our generic skip button click event).
* Verify `calypso_jpc_skipped_site_topic` event gets recorded (our specific site topic skip event).
* On the user type step, click the skip button.
* Verify `calypso_jpc_skip_button_click` event gets recorded (our generic skip button click event).
* Verify `calypso_jpc_skipped_user_type` event gets recorded (our specific user type skip event).

Fixes #31115 
Fixes #31784

jeffgolenski commented:
@tyxla Tested (after much local dev frustration, thanks for helping again). Looks great and works as expected. 

I'm curious about the data tracking though. Regarding the funnels I've created to see folks going from login / account creation all the way through plan purchase. Is there some way we can differentiate a "skip" from a "drop" in the flow?

tyxla commented:
Thanks Jeff! We're now recording some tracks events, in case you wish to test those:
* Type `localStorage.setItem( 'debug', 'calypso:analytics:tracks' )` in your browser console.
* On the site type step, click the skip button.
* Verify `calypso_jpc_skip_button_click` event gets recorded (our generic skip button click event).
* Verify `calypso_jpc_skipped_site_type` event gets recorded (our specific site type skip event).
* On the site topic step, click the skip button.
* Verify `calypso_jpc_skip_button_click` event gets recorded (our generic skip button click event).
* Verify `calypso_jpc_skipped_site_topic` event gets recorded (our specific site topic skip event).

fditrapani commented:
We recently ran some tests on WP.com with a skip button and found that it didn't have an impact on completion rates. Rather, people were picking up on the continue button being active at all times and being able to click that to move forward. I'd like to encourage us to move forward in that approach rather than adding more elements to the screen. More details here: p8Eqe3-xe-P2

shaunandrews commented:
> I'd like to encourage us to move forward in that approach rather than adding more elements to the screen

I can't +1 this enough. If we feel the need to add a skip link, then I think the bigger question is do we really need to have the step at all.

I could be on board with offering a "skip site setup" link on the first step which would skip all/most steps, but adding skip links to other steps is not a great solution.

tyxla commented:
Folks, would you like to suggest your thoughts about the rationale about adding or not adding a skip button in the related issue: #31115 - the thing is we already discussed this quite a bit in that issue.

tyxla commented:
I'm happy to close this PR if we conclude that Skip buttons aren't good, but last time I checked, we were agreeing on having them.

fditrapani commented:
Thanks for pointing out the issue @tyxla. We can move the discussion there. I wasn't aware it was happening elsewhere.

jeffgolenski commented:
@fditrapani @shaunandrews 

> I can't +1 this enough. If we feel the need to add a skip link, then I think the bigger question is do we really need to have the step at all.

I both agree and disagree here. In the context of Jetpack we eventually want to use the data collected here to influence a customized setup and overall jetpack experience. But in no way are these two inputs required to use Jetpack. That being said, if customers want to willingly give us this information so we can offer a better experience in the future, and learn more about them, great. But we should not be requiring this information or forcing them to give it to us. If anything, without a skip button here, they may be diluting our data just by clicking or entering the first thing they see. example: "I'm not any of this stuff, I'll just select business since its first on the list." I'd rather give them to ability to just skip it than give us incorrect info.

keoshi commented:
@fditrapani I took a quick look at that thread and saw no mentions of this; got a question for you:

> people were picking up on the continue button being active at all times

It seems like this test ran before we changed the buttons to be disabled by default. Don't know exactly when that happened, but could certainly affect the outcome if the big pink CTA isn't available.

fditrapani commented:
> Don't know exactly when that happened, but could certainly affect the outcome if the big pink CTA isn't available.

Yeah, I agree that could have an impact. The buttons were active during the test as far as I know.

fditrapani commented:
I also wanted to highlight the site type step converts at 92% according to @jeffgolenski last post. Curious why we would need a skip button there?

jeffgolenski commented:
> I also wanted to highlight the site type step converts at 92% according to @jeffgolenski last post. Curious why we would need a skip button there?

The site type > site topic converts at 92%, but the step before that, connection auth > site type is much much much lower. Coming in at 50% around the first week of insights. That's mostly what I'm concerned about. But I'm always working to optimize - 92% is not good enough for me, we can do better. if 12k people see the site type, why are ~1,000 people dropping off? We can save them! :-P

Also, we're taking into consideration other demographics than your typical end user. We have agencies and builders who set up a lot of sites on behalf of other people and I really don't want to get in their way or slow them down. I have friends who run agencies with over 50 Jetpack's installed. I would not want to be going through this onboarding every time. (+1 to shaun's skip setup entirely idea).

fditrapani commented:
> Also, we're taking into consideration other demographics than your typical end user. We have agencies and builders who set up a lot of sites on behalf of other people and I really don't want to get in their way or slow them down. I have friends who run agencies with over 50 Jetpack's installed. I would not want to be going through this onboarding every time. (+1 to shaun's skip setup entirely idea).

 Yeah, I like Shaun's idea there too. This is a change we want to make on the WP.com side too. 

> But I'm always working to optimize - 92% is not good enough for me, we can do better.
I'm not advocating for not testing but there comes a point where the effort to increase that number is way higher than what it's really worth and we need to make sure we're prioritizing the right work.

tyxla commented:
Sounds great! Let's get it code-reviewed first though, cc @Automattic/luna

tyxla commented:
I've rebased, adapted to the dark color scheme and added the skip button to the new User Type step.

This is ready for another design review. Also, code review has still not been done yet and I'd realy appreciate one. 🙌

tyxla commented:
@keoshi guess I missed to push the latest changes. Pushed now, would appreciate another 👁

tyxla commented:
Thanks @keoshi! Removed the focus box shadow in https://github.com/Automattic/wp-calypso/pull/31304/commits/47a64498acd3d7d59a2f56a42ff365c83a84f7b0: 

![](https://cldup.com/AeKjhrVonS.png)

jeffgolenski commented:
@keoshi @tyxla 

The skip buttons work great, Marin. Thank you

![Screen Shot 2019-03-25 at 4 58 40 PM](https://user-images.githubusercontent.com/214813/54953745-5929c180-4f1f-11e9-88c3-d1c01a9380df.png)


However, I ran a color accessibility test on this color link on top of that BG color and it fails both AA and AAA at its current size. Should we implement a lighter color var for the skip links and all the links that fall under the white boxes during out onboarding?

![Screen Shot 2019-03-25 at 4 59 49 PM](https://user-images.githubusercontent.com/214813/54953803-7e1e3480-4f1f-11e9-9653-1acb050d45eb.png)

tyxla commented:
Sounds good, but this also sounds like work for separate PR @jeffgolenski ? What I suggested here is using the same colors that we already use for the footer links in other places.

jeffgolenski commented:
Color ref: in https://github.com/Automattic/wp-calypso/issues/31784

Let's get the colors updated before we merge this skip link.

tyxla commented:
Sounds good @jeffgolenski and @keoshi! Updated colors from `-300` to `-200` for the footer links and the skip button:

Normal:
![](https://cldup.com/te28pM9hPT.png)

Focused/hover
![](https://cldup.com/pgOoEssrjt.png)

Let me know if this is good to go on the design side.

tyxla commented:
This still needs a code review, cc @Automattic/luna

jeffgolenski commented:
@tyxla @keoshi after the code review from luna, let's ship it. I'll spin up a funnel with the new tracks events to monitor its usage.

tyxla commented on client/jetpack-connect/skip-button.js:
```diff
@@ -0,0 +1,61 @@
+		this.props.recordTracksEvent( 'calypso_jpc_skip_button_click' );
+
+		if ( tracksEventName ) {
+			this.props.recordTracksEvent( tracksEventName );
```
Yeah, we used to have a single event, but @jeffgolenski requested 2 events specifically. I have no strong feelings either way, which is why I chosen to make his life easier 😉

tyxla commented:
Thank you, folks! I'm going to proceed and ship this when tests are green.

jeffgolenski commented:
You're a rockstar!
